packages/sns/lib/utils/snsUtils.ts

```diff
@@ -178,3 +193,35 @@ export async function getTopicArnByName(snsClient: SNSClient, topicName?: string
  */
 export const calculateOutgoingMessageSize = (message: unknown) =>
   sqsCalculateOutgoingMessageSize(message)
+
+const isTopicAlreadyExistWithDifferentTagsError = (error: unknown) =>
+  !!error &&
+  isError(error) &&
+  'Error' in error &&
+  !!error.Error &&
+  typeof error.Error === 'object' &&
+  'Code' in error.Error &&
+  'Message' in error.Error &&
+  typeof error.Error.Message === 'string' &&
+  error.Error.Code === 'InvalidParameter' &&
+  error.Error.Message.includes('already exists with different tags')
+
+/**
+ * Manually builds the ARN of a topic based on the current AWS account and the topic name.
+ * It follows the following pattern: arn:aws:sns:<region>:<account-id>:<topic-name>
+ * Doc -> https://docs.aws.amazon.com/IAM/latest/UserGuide/reference-arns.html
+ */
+const buildTopicArn = async (client: SNSClient, topicName: string) => {
+  const region =
+    typeof client.config.region === 'string' ? client.config.region : await client.config.region()
+
+  const stsClient = new STSClient({
```

Comment:
recreating client on every response sounds like a lot of overhead, I'd suggest to pass it as a dependency instead

Reply:
Do you mean as a dependency for users? the idea was to hide its usage by using SNS config parameters although, I am not really sure 🤔.

What do you mean by recreating it in every response? I think it is done just once on init right?

Reply:
Oh I got it, you mean to the method, would be fine to create it on publisher/consumer creation and pass it as parameter?

Reply:
wouldn't it be done once per buildTopicArn, which realistically means once for every topic?

Reply:
yeah, same as we pass sqsClient and snsClient

Reply:
But as a dependency passed by the user, or do you think it is a good idea to build it internally base on SNS config? Advantage of not requesting it is that the API won’t change

Reply:
(Btw sorry, I am already a bit slow 😓)

Reply:
I would pass by user, would be less surprising than globally cached dependency. but we can also make an optional parameter and build automatically if not passed